src/Inventory/Asset/Software.php

```diff
@@ -726,7 +733,7 @@ private function storeSoftware()
                 //set create date
                 $stmt_columns['date_creation'] = $_SESSION["glpi_currenttime"];
 
-                if ($stmt === null) {
+                if ($stmt === null || $stmt_types === null || strlen($stmt_types) != count($stmt_columns)) {
```

Comment:
I think it's wrong to silentely check columns count. If columns count is wrong, there is a bug that must be fixed.